test/integration/bastion/bastion_test.go

```diff
@@ -0,0 +1,640 @@
+	if err := waitForOperation(ctx, project, computeService, routerOp); err != nil {
+		return err
+	}
+
+	subnetOp, err := computeService.Subnetworks.Delete(project, *region, subnetName).Context(ctx).Do()
+	if err != nil {
+		return err
+	}
+
+	logger.Info("Waiting until subnet is deleted...", "subnet ", subnetName)
+	if err := waitForOperation(ctx, project, computeService, subnetOp); err != nil {
+		return err
+	}
+
+	networkOp, err := computeService.Networks.Delete(project, networkName).Context(ctx).Do()
+	if err != nil {
+		return err
+	}
+
+	logger.Info("Waiting until network is deleted...", "network ", networkName)
+
+	err = waitForOperation(ctx, project, computeService, networkOp)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func waitForOperation(ctx context.Context, project string, computeService *compute.Service, op *compute.Operation) error {
+	return wait.PollUntil(5*time.Second, func() (bool, error) {
+		var (
+			currentOp *compute.Operation
+			err       error
+		)
+
+		if op.Region != "" {
+			region := getResourceNameFromSelfLink(op.Region)
+			currentOp, err = computeService.RegionOperations.Get(project, region, op.Name).Context(ctx).Do()
+		} else {
+			currentOp, err = computeService.GlobalOperations.Get(project, op.Name).Context(ctx).Do()
+		}
+
+		if err != nil {
+			return false, err
+		}
+		return currentOp.Status == "DONE", nil
+	}, ctx.Done())
+}
+
+func getResourceNameFromSelfLink(link string) string {
+	parts := strings.Split(link, "/")
+	return parts[len(parts)-1]
+}
+
+func createBastion(cluster *controller.Cluster, name string, project string) (*extensionsv1alpha1.Bastion, *bastionctrl.Options) {
+	bastion := &extensionsv1alpha1.Bastion{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name + "-bastion",
+			Namespace: name,
+		},
+		Spec: extensionsv1alpha1.BastionSpec{
+			DefaultSpec: extensionsv1alpha1.DefaultSpec{
+				Type: "gcp",
+			},
+			UserData: []byte(userDataConst),
+			Ingress: []extensionsv1alpha1.BastionIngressPolicy{
+				{IPBlock: networkingv1.IPBlock{
+					CIDR: myPublicIP,
+				}},
+			},
+		},
+	}
+
+	options, err := bastionctrl.DetermineOptions(bastion, cluster, project)
+	Expect(err).NotTo(HaveOccurred())
+
+	return bastion, options
+}
+
+func createInfrastructureConfig() *gcpv1alpha1.InfrastructureConfig {
+	return &gcpv1alpha1.InfrastructureConfig{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: gcpv1alpha1.SchemeGroupVersion.String(),
+			Kind:       "InfrastructureConfig",
+		},
+		Networks: gcpv1alpha1.NetworkConfig{
+			Workers: workersSubnetCIDR,
+		},
+	}
+}
+
+func createShootStruct(infrastructureConfig []byte) *gardencorev1beta1.Shoot {
+
+	shoot := &gardencorev1beta1.Shoot{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "core.gardener.cloud/v1beta1",
+			Kind:       "Shoot",
+		},
+		Spec: gardencorev1beta1.ShootSpec{
+			Region: *region,
+			Provider: gardencorev1beta1.Provider{
+				InfrastructureConfig: &runtime.RawExtension{
+					Raw: []byte(infrastructureConfig),
```

Comment:
The infrastructureConfig is already a `[]byte`, no need to cast additionally.

```suggestion
					Raw: infrastructureConfig,
```

Reply:
fixed 783eb84